pinot-server/src/main/java/org/apache/pinot/server/request/ScheduledRequestHandler.java

```diff
@@ -51,20 +49,14 @@ public ScheduledRequestHandler(QueryScheduler queryScheduler, ServerMetrics serv
   }
 
   @Override
-  public ListenableFuture<byte[]> processRequest(ChannelHandlerContext channelHandlerContext, ByteBuf request) {
+  public ListenableFuture<byte[]> processRequest(byte[] request) {
     long queryArrivalTimeMs = System.currentTimeMillis();
     serverMetrics.addMeteredGlobalValue(ServerMeter.QUERIES, 1);
 
-    LOGGER.debug("Processing request : {}", request);
-
-    byte[] byteArray = new byte[request.readableBytes()];
-    request.readBytes(byteArray);
     SerDe serDe = new SerDe(new TCompactProtocol.Factory());
-    final InstanceRequest instanceRequest = new InstanceRequest();
-
-    if (!serDe.deserialize(instanceRequest, byteArray)) {
-      LOGGER.error("Failed to deserialize query request from broker ip: {}",
-          ((InetSocketAddress) channelHandlerContext.channel().remoteAddress()).getAddress().getHostAddress());
+    InstanceRequest instanceRequest = new InstanceRequest();
+    if (!serDe.deserialize(instanceRequest, request)) {
+      LOGGER.error("Failed to deserialize query request: {}", BytesUtils.toHexString(request));
```

Comment:
Adding `BytesUtils.toHexString(request)` seems not to add much value on debugging?

Reply:
This should not happen. While if it happens, we want to know the bytes of the request in order to debug.